lnst/Common/Parameters.py

```diff
@@ -179,6 +180,53 @@ def type_check(self, value):
         return value
 
 
+class NetworkParam(Param):
+    """Network Param
+
+    This parameter is used to specify the IP networks a test will have
+     interfaces on. Using it allows multiple simultaneous LNST sessions on a shared broadcast domain.
+
+     It is used with physical or created logical interfaces (vlans, bonds, etc) have
+     addresses on external networks.
```

Comment:
I don't understand this sentence. What is meant by external networks?

Reply:
that should probably say "that have addresses". ]

Yeah I basically want to say you only need to use it when the addresses you pick are exposed to the network outside the nic. Thats what I mean by "external network" here.  I'll see if i can reword it better, if you have any suggestions I'm all ears.

Reply:
i think you're overcomplicating the docstring here. this is a `lnst.Common.Parameters` module class... which means it can be used universally for any ip network use case a tester can imagine.

So IMO the docstring should primarily focus on describing the fact that this parameter class accepts strings formatted in a way that describe an IPv4 or IPv6 network.

And only after that provide an example of a potential use case where we use this specifically to configure ip addresses that our ENRT recipes use on the baremetal nics that are connected to shared networks to avoid conflicts.

... actually, this last part about how the ENRT test set uses this may be more fit to include as documentation in the ENRT package instead of here in the `Common` module... and then maybe link it from here as an example of a potential use?

Reply:
Re did the interface_address and NetworkParam docstrings let me know what you think.